cpp/core/utils/ObjectStore.cc

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ObjectStore.h"
+#include <iostream>
+
+gluten::ObjectStore::~ObjectStore() {
+  // destructing in reversed order (the last added object destructed first)
+  const std::lock_guard<std::mutex> lock(mtx_);
+  for (auto itr = aliveObjectHandles_.rbegin(); itr != aliveObjectHandles_.rend(); itr++) {
+    ResourceHandle handle = *itr;
+    if (store_.lookup(handle) == nullptr) {
+      std::cerr << "Fatal: resource handle " + std::to_string(handle) + " not found in store." << std::endl;
```

Comment:
I think this case it not acceptable and you use Fatal too, so use GLUTEN_FAIL or anyway to throw exception to kill task?

Reply:
No, it's in destructor. We'd avoid throwing from destructor.